### ISSUE TITLE
sitemaps-1-sitemap.xml in robots.txt

### ISSUE THREAD

**Issue by brandondeweese** [nystudio107/craft-seomatic#731]
Hi, is there a reason the sitemap URL in the robots.txt is "sitemaps-1-sitemap.xml" and not"sitemap.xml"?

**Comment by OwenMelbz** [comment]
We've found out that "sitemap.xml" is just a friendly alias for search engines/tools that cannot look for sitemaps in custom locations.

Would be nice to turn this off/on or enforce the full url via a redirect we actually have this redirect set up

```
redirect 301 /sitemap.xml /sitemaps-1-sitemap.xml
```

**Update:**

Until/Unless this addressed upstream this is our "solution" incase anybody finds it helpful to them.

We've got a new class called "SitemapRedirects.php"

```php
<?php

namespace modules\Storm\Middleware;

use Craft;
use nystudio107\seomatic\services\Helper;

class SitemapRedirects
{
    public static function handle()
    {
        if (Craft::$app->request->getPathInfo() === 'sitemap.xml') {
            header('Location: ' . Helper::sitemapIndexForSiteId());
            exit;
        }
    }
}
```

Then we've got our base module which hooks into it

```php
<?php

namespace modules\Storm;

use yii\base\Event;
use yii\base\Module;
use yii\web\Application;

class Storm extends Module
{
    public function init()
    {
        Craft::setAlias('@modules/Storm', __DIR__);

        Yii::configure($this, require(__DIR__ . '/config.php'));

        require(__DIR__ . '/helpers.php');

        Event::on(Application::class, Application::EVENT_BEFORE_REQUEST, function () {
            SitemapRedirects::handle();
        });

        parent::init();
    }
}
```

**Comment by khalwat** [comment]
Thanks @OwenMelbz very helpful!

**Comment by khalwat** [comment]
> Would be nice to turn this off/on or enforce the full url via a redirect we actually have this redirect set up

I'm not sure this is actually a problem. If there's some evidence that it is, SEOmatic could implement that redirect itself.

**Comment by brandondeweese** [comment]
Is there any adverse effects to directing it to the sitemap.xml manually?

Services like SEMrush, mark this is as an error that needs to be fixed...

**Comment by khalwat** [comment]
@brandondeweese Show me the error the point out? A screenshot would be wonderful.

**Comment by brandondeweese** [comment]
![Screen Shot 2020-08-28 at 6 33 16 AM](https://user-images.githubusercontent.com/18017918/91566361-67d88d00-e8f8-11ea-96e8-dfe650494b72.png)

**Comment by khalwat** [comment]
@brandondeweese Sounds like there might be an issue with their tool? Because SEOmatic definitely provides a sitemap, visible at `/sitemap.xml` 

Can you give me your live URL to look at?

**Comment by brandondeweese** [comment]
I think it's directly linked to the robots.txt file. That's why I was originally asking this because my sitemap.xml file is there but in the robots.txt it was directing to this: sitemaps-1-sitemap.xml

https://valleywestlandscapes.com/sitemap.xml

I've updated this since this SEMrush report but this isn't my report, it was from my client's own SEO contractor. I haven't asked him to re-run the report since I've made the update.
https://valleywestlandscapes.com/robots.txt
